controllers/agentmachine_controller_test.go

```diff
@@ -209,7 +209,8 @@ var _ = Describe("agentmachine reconcile", func() {
 	)
 
 	BeforeEach(func() {
-		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		agentMachine := &capiproviderv1.AgentMachine{}
+		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).WithStatusSubresource(agentMachine).Build()
```

Comment:
For history, had to add WithStatusSubresource to this because of https://github.com/kubernetes-sigs/controller-runtime/issues/2362 which prevents the status of a resource from being updated during unit tests, causing them to fail.
Change ref: https://github.com/evryfs/github-actions-runner-operator/commit/165da33dd0cdd7ab85dff76f9bc6b851b8bb4a1f